DownFall/Views/Header/Header.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  Header.swift
+//  DownFall
+//
+//  Created by William Katz on 3/3/19.
+//  Copyright © 2019 William Katz LLC. All rights reserved.
+//
+
+import SpriteKit
+
+class Header: SKSpriteNode {
+    
+    static func build(color: UIColor,
+                      size: CGSize) -> Header {
+        let header = Header(texture: SKTexture(imageNamed: "header"), color: color, size: size)
+        let setting = SKSpriteNode(imageNamed: "setting")
+        setting.name = "setting"
+        let settingX = header.frame.maxX - setting.frame.width
+        setting.position = CGPoint(x: settingX, y: size.height/2 - setting.frame.height)
+        setting.zPosition = 3
+        
+        header.addChild(setting)
+        header.isUserInteractionEnabled = true
+        return header
+    }
+    
+    override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent?) {
```

Comment:
Since this code is duplicated in a bunch of places, we could consider creating a protocol for all objects to conform to that are interested in touch events.  

```protocol TouchEventListener```, it could provide default implementation for certain touch events and then call another method defined on the protocol like ```nodesTouched(nodes:)```

Reply:
The other option is to create a a global helper that given a node and touch events can tell you all the nodes that were touched